ntoskrnl/ps/query.c

```diff
@@ -49,6 +49,592 @@ PsReferenceProcessFilePointer(IN PEPROCESS Process,
     return Section ? STATUS_SUCCESS : STATUS_UNSUCCESSFUL;
 }
 
+/**
+ * @brief
+ * Dumps the name of a process information class.
+ *
+ * @param[in] ProcessInformationClass
+ * The enumeration code value of a process information class,
+ * of which the name of it is returned as a string.
+ *
+ * @return
+ * Returns a pointer to a string of the process information
+ * class name.
+ */
+static
+PWSTR
+PspDumpProcessInfoClassName(
+    _In_ PROCESSINFOCLASS ProcessInformationClass)
+{
+    switch (ProcessInformationClass)
+    {
+        case ProcessBasicInformation:
+        {
+            return L"ProcessBasicInformation";
```

Comment:
This would probably worth drive with a macro.
```c
#define DBG_PROCESS_INFO_CLASS(class) \
    case class: return L ## class
...
    switch (class)
    {
        DBG_PROCESS_INFO_CLASS(ProcessBasicInformation);
        DBG_PROCESS_INFO_CLASS(ProcessWhatever1);
        ...
    }
#undef DBG_PROCESS_INFO_CLASS
```

Reply:
I used a constant array for table lookup instead.